packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -401,15 +402,25 @@ class _SearchAnchorState extends State<SearchAnchor> {
 
   @override
   void dispose() {
-    super.dispose();
     widget.searchController?._detach(this);
     _internalSearchController?._detach(this);
-    _internalSearchController?.dispose();
-  }
+    final bool usingExternalController = widget.searchController != null;
+    if (_viewIsOpen) {
+      _route?._dismiss(
+        disposeController: !usingExternalController,
+      );
+      if (usingExternalController){
```

Comment:
```suggestion
      if (usingExternalController) {
```

Reply:
Thanks for the review and suggestions. This issue was quite tricky.